File: OOPPractice/ex2_3_2.rb
class ObscuringReferences
  attr_reader :data

  def initialize(data)
    @data = data
  end

  # dataの構造に依存したメソッド
  def diameters
    p data
    data.collect { |cell|
      cell[0] + (cell[1] * 2)
    }
  end
end

data = [[622, 20], [622, 23], [559, 30], [559, 40]]
p ObscuringReferences.new(data).diameters

File: OOPPractice/ex2_3_2.php
<?php

class ObscuringReferences
{
    private $data;

    function __construct($data)
    {
        $this->data = $data;
    }

    function data()
    {
        return $this->data;
    }

    // this->dataの構造に依存したメソッド
    function diameters()
    {
        $data = [];
        foreach ($this->data() as $cell) {
            $data[] = $cell[0] + ($cell[1] * 2);
        }
        return $data;
    }
}

$data = [[622, 20], [622, 23], [559, 30], [559, 40]];
$or = new ObscuringReferences($data);
print_r($or->diameters());
